pdm/project/core.py

```diff
@@ -298,7 +299,15 @@ def sources(self) -> list[Source]:
                     "name": "pypi",
                 },
             )
-        return sources
+        expanded_sources: list[Source] = [
+            Source(
+                url=expand_env_vars_in_auth(s["url"]),
+                verify_ssl=s["verify_ssl"],
+                name=s["name"],
```

Comment:
Since the `Source` `TypedDict` has all three keys required, should defaults for the others be provided? Perhaps `False` and `""` for `verify_ssl` and `name` respectively.

Reply:
No, the related code is here: https://github.com/pdm-project/pdm/blob/22d8bd0ac319ec7dcc3534abb596f2520b4838f2/pdm/utils.py#L80-L96

`very_ssl` is derived from the URL if not given

Reply:
Yes I think it is OK